domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClient.java

```diff
@@ -497,6 +502,27 @@ private void init() throws Exception {
     }
 
     client = createPgPool(vertx, postgreSQLClientConfig);
+
+    activeConnections = new CopyOnWriteArraySet<>();
+
+    vertx.setPeriodic(CHECK_FOR_QUERY_TIMEOUT_INTERVAL, timerId -> {
+      for (SQLConnection conn : activeConnections) {
+        long elapsedTime = System.currentTimeMillis() - conn.acquiringTime;
+        if (elapsedTime > conn.executionTimeLimit) {
+          conn.conn.cancelRequest(ar -> {
```

Comment:
How do you ensure that the timeout of a first query doesn't cancel a second query running on the same connection?
```
SELECT 1, pg_sleep(2);  // runs 2 seconds, use timeout = 2500 ms
SELECT 2, pg_sleep(2);  // runs 2 seconds, use timeout = 2500 ms
```
Can you write a unit test for this?

Reply:
Should it be the case? If the connection is retrieved from pool, it means that request is started. The initial problem id that request should be limited with certain time. So the limitation of queries within the same connection in my opinion makes sense.

Reply:
Like this:  
  ```
@Test
  public void selectWithTimeout(TestContext context) {
      PostgresClient client = postgresClient();
      client.getSQLConnection(2500, conn -> {
        client.selectSingle(conn, "SELECT 1, pg_sleep(2);", context.asyncAssertSuccess());
        client.selectSingle(conn, "SELECT 1, pg_sleep(2);", context.asyncAssertFailure());
      });
  }
```

Reply:
Fixed